library/classes/rulesets/PQRS/reports/PQRS_0008/Exclusion1.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * PQRS Measure 0008 -- Exclusion 
+ *
+ * Copyright (C) 2015 - 2017      Suncoast Connection
+  * 
+ * LICENSE: This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0
+ * See the Mozilla Public License for more details. 
+ * If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ * 
+ * @author  Art Eaton <art@suncoastconnection.com>
+ * @author  Bryan lee <bryan@suncoastconnection.com>
+ * @package LibreHealthEHR 
+ * @link    http://suncoastconnection.com
+ * @link    http://librehealth.io
+ *
+ * Please support this product by sharing your changes with the LibreHealth.io community.
+ */
+
+class PQRS_0008_Exclusion1 extends PQRSFilter
+{
+    public function getTitle() 
+    {
+        return "Exclusion";
+    }
+    
+    public function test( PQRSPatient $patient, $beginDate, $endDate )
+    {
+$query =
+"SELECT COUNT(b1.code) as count".  
+" FROM billing AS b1".
+" JOIN form_encounter AS fe ON (b1.encounter = fe.encounter)".
+" WHERE b1.pid = ? ".
+" AND fe.date BETWEEN '".$beginDate."' AND '".$endDate."' ".
+" AND b1.code ='G8451';";
+
+$result = sqlFetchArray(sqlStatementNoLog($query, array($patient->id)));
```

Comment:
When it is more than one code in a list for an exception, we make a separate join.  When it is a single code, we just use the code.  It may LOOK like the same exception is made, but there are lots of codes that mean exactly the same thing...and there is just a lot to that I am still addressing.  The major salient point is this:  We have tried a number of things, but the way stuff is currently written...by rigorous testing, is by FAR the fastest.  Some of this flies in the face of commonly accepted knowledge.  We have tweaked this stuff (with a very few exceptions) to run all of these measures against thousands of patients in just a few minutes.  The fancier-seeming original CQM stuff with "helpers" and other functions can take hours to run just one measure.  The good thing we have for test-driving this is the massive log output stuff Bryan put in.  It allows you to tail the log and watch how long each measure takes to run.  Under that scheme, once this is all running under Libre and we have taken care of getting rid of and actual security risks or unwanted features, we can do a lot more experimenting and maybe do some fancier stuff.  This year we only have about 250 QA measures, and about 100 pre-measures, and only another hundred or so IA and clinical improvement thingies.  Last year we had all the group measures too (about double the size of this).  The standards for the QA part are here: http://suncoastconnection.com/standards/index.php/